File: src/main/java/vocadb/notification/reader/client/query/SongSortRule.java
package vocadb.notification.reader.client.query;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum SongSortRule {
    @JsonProperty("None")
    NONE,
    @JsonProperty("Name")
    NAME,
    @JsonProperty("AdditionDate")
    ADDITION_DATE,
    @JsonProperty("FavoritedTimes")
    FAVORITED_TIMES,
    @JsonProperty("RatingScore")
    RATING_SCORE;
}

File: src/main/java/vocadb/notification/reader/client/api/enums/SongSortRule.kt
package vocadb.notification.reader.client.api.enums

import com.fasterxml.jackson.annotation.JsonValue

enum class SongSortRule(@JsonValue val value: String) {
    NONE("None"),
    NAME("Name"),
    ADDITION_DATE("AdditionDate"),
    FAVORITED_TIMES("FavoritedTimes"),
    RATING_SCORE("RatingScore")
}
